src/java.base/share/classes/java/io/StringReader.java

```diff
@@ -207,7 +177,11 @@ public void reset() throws IOException {
      */
     public void close() {
         synchronized (lock) {
-            str = null;
+            try {
+                r.close();
+            } catch (IOException e) {
+                throw new UncheckedIOException(e);
```

Comment:
In fact, the implementation of `close` in the original class will never throw `IOException`, but unfortunately `Reader.close()` declares `throws IOException`.